File: beejx_backend/services/weather_service.py
import requests
import os
from dotenv import load_dotenv

load_dotenv()

API_KEY = os.getenv("OPENWEATHER_API_KEY", "YOUR_OWM_KEY")
BASE_URL = "https://api.openweathermap.org/data/2.5/weather"

def fetch_weather(lat: float, lon: float):
    try:
        if API_KEY == "YOUR_OWM_KEY":
            # Return mock data if no key provided
            return {
                "temp": 28,
                "condition": "Sunny (Mock)",
                "humidity": 65,
                "wind": 12,
                "location": "Uttarakhand (Mock)"
            }

        params = {
            "lat": lat,
            "lon": lon,
            "appid": API_KEY,
            "units": "metric"
        }
        response = requests.get(BASE_URL, params=params)
        data = response.json()

        if response.status_code == 200:
            return {
                "temp": int(data["main"]["temp"]),
                "condition": data["weather"][0]["main"],
                "humidity": data["main"]["humidity"],
                "wind": data["wind"]["speed"],
                "location": data["name"]
            }
        else:
            return {"error": "Unable to fetch weather"}
            
    except Exception as e:
        print(f"Weather Error: {e}")
        return {"error": str(e)}

File: lib/services/weather_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../utils/constants.dart';

class WeatherService {
  Future<Map<String, dynamic>> getWeather(double lat, double lon) async {
    try {
      // Use configured endpoint (http://192.../api/v1/weather)
      final response = await http.get(
        Uri.parse('${ApiConstants.weatherEndpoint}?lat=$lat&lon=$lon'),
      );

      if (response.statusCode == 200) {
        return jsonDecode(response.body);
      } else {
        throw Exception('Failed to load weather');
      }
    } catch (e) {
      // Return fallback data if API fails
      return {
        "temp": 28,
        "condition": "Sunny (Offline)",
        "humidity": 65,
        "wind": 12,
        "location": "Uttarakhand"
      };
    }
  }
}
